core/src/main/java/com/datastax/oss/driver/internal/core/metadata/AddNodeRefresh.java

```diff
@@ -54,7 +54,7 @@ public Result compute(
       // an addition, even though the host_id hasn't changed :(
       // Update the existing instance and emit an UP event to trigger a pool reconnection.
       if (!existing.getEndPoint().equals(newNodeInfo.getEndPoint())) {
-        copyInfos(newNodeInfo, ((DefaultNode) existing), null, context);
```

Comment:
Was the problem that we were passing null and because of that there was no recalculation of this condition:
https://github.com/datastax/java-driver/pull/1436/files#diff-7a875990776d2f9525a97a8561fa1cfaL61
?

Reply:
Yes correct, although when the node is new, you hit line 43, not 57. But in both cases, `newNodeInfo` contains the new node's tokens, but that information wasn't being copied to `existing` or `newNode`.